### PR TITLE
use spotahome redis-operator

### PR DESCRIPTION
**Pull request by yxxhero**
Signed-off-by: yxxhero <aiopsclub@163.com>

use spotahome redis-operator

### CODE REVIEW

File: .github/workflows/build-dev.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     steps:
     - uses: actions/setup-go@v2
       with:
-        go-version: 1.16
+        go-version: 1.17
```
Comment (cndoit18) [inline .github/workflows/build-dev.yml:15]: Can you create a new PR to do golang upgrade?

File: controllers/goharbor/harborcluster/status.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ import (
 // status is designed to track the status and conditions of the deploying Harbor cluster.
 type status struct {
 	client.Client
-	log     logr.Logger
+	log     *logr.Logger
```
Comment (cndoit18) [inline controllers/goharbor/harborcluster/status.go:35]: Why is it to change to a pointer

File: pkg/factories/logger/logger.go [hunk 3]
```diff
@@ -12,7 +12,7 @@ var loggerContext = "logger"
 func Get(ctx context.Context) logr.Logger {
 	l := ctx.Value(&loggerContext)
 	if l == nil {
-		return ctrl.Log
+		return ctrl.Log.WithName("")
```
Comment (cndoit18) [inline pkg/factories/logger/logger.go:15]: I don't understand this change

File: .github/workflows/build-dev.yml (hunk 1 above)
Comment (yxxhero) [reply .github/workflows/build-dev.yml:15]: This PR need go1.17.

File: pkg/factories/logger/logger.go (hunk 3 above)
Comment (yxxhero) [reply pkg/factories/logger/logger.go:15]: https://github.com/go-logr/logr is upgrade. Logger change to struct from interface.

File: controllers/goharbor/harborcluster/status.go (hunk 2 above)
Comment (yxxhero) [reply controllers/goharbor/harborcluster/status.go:35]: https://github.com/go-logr/logr is upgrade. Logger change to struct from interface.

File: pkg/factories/logger/logger.go (hunk 3 above)
Comment (bitsf) [inline pkg/factories/logger/logger.go:15]: this change seems useless

**Review by cndoit18 — APPROVED** [review]
LGTM

**Review by bitsf — APPROVED** [review]

**Pull request merged**